caddyhttp/browse/browse.go

```diff
@@ -291,18 +293,13 @@ func isSymlinkTargetDir(f os.FileInfo, urlPath string, config *Config) bool {
 	if !isSymlink(f) {
 		return false
 	}
-	fullPath := func(fileName string) string {
-		fullPath := filepath.Join(string(config.Fs.Root.(http.Dir)), urlPath, fileName)
-		return filepath.Clean(fullPath)
-	}
-	target, err := os.Readlink(fullPath(f.Name()))
-	if err != nil {
-		return false
-	}
-	targetInfo, err := os.Lstat(fullPath(target))
+
+	fullPath := filepath.Join(string(config.Fs.Root.(http.Dir)), urlPath, f.Name())
+	targetInfo, err := os.Stat(fullPath)
```

Comment:
This seems to work:
```go
func isSymlinkTargetDir(f os.FileInfo, urlPath string, config *Config) bool {
	if !isSymlink(f) {
		return false
	}

	target, err := config.Fs.Root.Open(filepath.Join(urlPath, f.Name()))
	if err != nil {
		return false
	}
	defer target.Close()
	targetInto, err := target.Stat()
	if err != nil {
		return false
	}

	return targetInto.IsDir()
}
```

That should make sure the symlink we are following is located inside `config.Fs.Root`.

With the original code im not really sure what an attacker could do, maybe stat files outside the root using a relative urlPath joined with the symlink target path.